src/mlpack/methods/reinforcement_learning/replay/prioritized_replay.hpp

```diff
@@ -0,0 +1,269 @@
+  {
+    arma::ucolvec idxes(batchSize);
+    double totalSum = idxSum.sum(0, (full ? capacity : position));
+    double sumPerRange = totalSum / batchSize;
+    for (size_t bt = 0; bt < batchSize; bt ++) {
+      double mass = arma::randu() * sumPerRange + bt * sumPerRange;
+      size_t idx = idxSum.findPrefixSum(mass);
+      idxes[bt] = idx;
+    }
+    return idxes;
+  }
+
+  /**
+   * Sample some experience according to their priorities.
+   *
+   * @param sampledStates Sampled encoded states.
+   * @param sampledActions Sampled actions.
+   * @param sampledRewards Sampled rewards.
+   * @param sampledNextStates Sampled encoded next states.
+   * @param isTerminal Indicate whether corresponding next state is terminal
+   *        state.
+   * @param sampledIndices Sampled indices
+   * @param weights Corresponding weight for updating the loss
+   */
+  void Sample(arma::mat& sampledStates,
+              arma::icolvec& sampledActions,
+              arma::colvec& sampledRewards,
+              arma::mat& sampledNextStates,
+              arma::icolvec& isTerminal,
+              arma::ucolvec& sampledIndices,
+              arma::rowvec& weights)
+  {
+    size_t upperBound = full ? capacity : position;
+
+    sampledIndices = sampleProportional();
+    betaAnneal();
+
+    sampledStates = states.cols(sampledIndices);
+    sampledActions = actions.elem(sampledIndices);
+    sampledRewards = rewards.elem(sampledIndices);
+    sampledNextStates = nextStates.cols(sampledIndices);
+    isTerminal = this->isTerminal.elem(sampledIndices);
+
+//  calculate the weights of sampled transitions
+
+    size_t num_sample = full ? capacity : position;
+    weights = arma::rowvec(sampledIndices.n_rows);
+
+    for (size_t i = 0; i < sampledIndices.n_rows; ++ i)
+    {
+      double p_sample = idxSum.get(sampledIndices[i]) / idxSum.sum();
+      weights[i] = pow(num_sample * p_sample, -beta);
+    }
+    weights /= weights.max();
+  }
+
+  /**
+   * Update priorities of sampled transitions.
+   *
+   * @param indices The indices of sample to be updated.
+   * @param priorities Their corresponding priorities.
+   */
+  void update_priorities(arma::ucolvec& indices, arma::colvec& priorities)
+  {
+      arma::colvec alphaPri(indices.n_rows);
+      for (size_t i = 0; i < indices.n_rows; ++i)
+      {
+        alphaPri = alpha * priorities[i];
+        max_priority = std::max(max_priority, priorities[i]);
+      }
+      idxSum.batchUpdate(indices, alphaPri);
+  }
+
+  /**
+   * Get the number of transitions in the memory.
+   *
+   * @return Actual used memory size
+   */
+  const size_t& Size()
+  {
+    return full ? capacity : position;
+  }
+
+  /**
+   *  Annealing the beta
+   */
+  void betaAnneal()
+  {
+    beta = beta + (1 - initial_beta) * 1.0 / replay_beta_iters;
+  }
+
+private:
+  //! How much prioritization is used.
+  //  (0 - no prioritization, 1 - full prioritization)
+  double alpha;
+
+  double max_priority;
```

Comment:
Please avoid underscore in the names of variables. I would recommend going through [style guidelines](https://github.com/mlpack/mlpack/wiki/DesignGuidelines) once.

Reply:
Hi @robotcator, I suppose you were mentioning me here, but somehow accidentally tagged @mirraaj. Anyway in this case, I would say either is fine. You may use whatever fits you best. It hardly makes any difference.
Though I would ask @rcurtin @zoq If they have any preference in this regard and would like to pick one for future uses?